ESPLoader.js

```diff
@@ -151,23 +155,31 @@ class ESPLoader {
 
         if (wait_response) {
             try {
-                var p = await this.transport.read({timeout: timeout});
-                //console.log("Response " + p);
-                const resp = p[0];
-                const op_ret = p[1];
-                const len_ret = this._bytearray_to_short(p[2], p[3]);
-                const val = this._bytearray_to_int(p[4], p[5], p[6], p[7]);
-                //console.log("Resp "+resp + " " + op_ret + " " + len_ret + " " + val );
-                const data = p.slice(8);
-                if (op == null || op_ret == op) {
-                    return [val, data];
-                } else {
-                    throw new ESPError("invalid response");
+                // Check up-to next 100 packets for valid response packet
+                for (let i=0 ; i<100 ; i++) {
+                    var p = await this.transport.read({timeout: timeout});
+                    //console.log("Response " + p);
+                    const resp = p[0];
+                    const op_ret = p[1];
+                    const len_ret = this._bytearray_to_short(p[2], p[3]);
+                    const val = this._bytearray_to_int(p[4], p[5], p[6], p[7]);
+                    //console.log("Resp "+resp + " " + op_ret + " " + len_ret + " " + val );
+                    const data = p.slice(8);
+                    if (resp == 1) {
+                        if (op == null || op_ret == op) {
+                            return [val, data];
+                        } else if (data[0] != 0 && data[1] == this.ROM_INVALID_RECV_MSG) {
+                            await this.flush_input();
+                            throw new ESPError("unsupported command error");
+                        }
+                    }
                 }
+                throw new ESPError("invalid response")
             } catch(e) {
                 if (e instanceof TimeoutError) {
                     throw(e);
                 }
+                console.log(e);
             }
```

Comment:
This will now catch "invalid response", that seems wrong. Should we always raise?

Reply:
Actually, yes, that was pretty dumb of me!